Environment Variables passing to Forge

I'm experiencing errors from Harbor that relate to processing of environment variables:

```
$ harbor provision
  INFO    Start finding the server.
  INFO    Finding the associated site.
  INFO    Processing update of environment variables.
[2024-04-11 12:24:08] development.ERROR: Undefined array key 1 {"exception":"[object] (ErrorException(code: 0): Undefined array key 1 at /composer/vendor/mehrancodes/laravel-harbor/app/Actions/MergeEnvironmentVariables.php:53)
[stacktrace]
#0 /composer/vendor/laravel-zero/foundation/src/Illuminate/Foundation/Bootstrap/HandleExceptions.php(255): Illuminate\\Foundation\\Bootstrap\\HandleExceptions->handleError()
#1 /composer/vendor/mehrancodes/laravel-harbor/app/Actions/MergeEnvironmentVariables.php(53): Illuminate\\Foundation\\Bootstrap\\HandleExceptions->Illuminate\\Foundation\\Bootstrap\\{closure}()
#2 /composer/vendor/mehrancodes/laravel-harbor/app/Actions/MergeEnvironmentVariables.php(31): App\\Actions\\MergeEnvironmentVariables->searchReplaceExistingVariables()
#3 /composer/vendor/lorisleiva/laravel-actions/src/Concerns/AsObject.php(24): App\\Actions\\MergeEnvironmentVariables->handle()
#4 /composer/vendor/mehrancodes/laravel-harbor/app/Services/Forge/Pipeline/UpdateEnvironmentVariables.php(47): App\\Actions\\MergeEnvironmentVariables::run()
#5 /composer/vendor/mehrancodes/laravel-harbor/app/Services/Forge/Pipeline/UpdateEnvironmentVariables.php(36): App\\Services\\Forge\\Pipeline\\UpdateEnvironmentVariables->getBothEnvsMerged()
#6 /composer/vendor/illuminate/pipeline/Pipeline.php(184): App\\Services\\Forge\\Pipeline\\UpdateEnvironmentVariables->__invoke()
#7 /composer/vendor/mehrancodes/laravel-harbor/app/Services/Forge/Pipeline/EnableQuickDeploy.php(27): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#8 /composer/vendor/illuminate/pipeline/Pipeline.php(184): App\\Services\\Forge\\Pipeline\\EnableQuickDeploy->__invoke()
#9 /composer/vendor/mehrancodes/laravel-harbor/app/Services/Forge/Pipeline/ObtainLetsEncryptCertification.php(28): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#10 /composer/vendor/illuminate/pipeline/Pipeline.php(184): App\\Services\\Forge\\Pipeline\\ObtainLetsEncryptCertification->__invoke()
#11 /composer/vendor/mehrancodes/laravel-harbor/app/Services/Forge/Pipeline/InstallGitRepository.php(27): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#12 /composer/vendor/illuminate/pipeline/Pipeline.php(184): App\\Services\\Forge\\Pipeline\\InstallGitRepository->__invoke()
#13 /composer/vendor/mehrancodes/laravel-harbor/app/Services/Forge/Pipeline/CreateDatabase.php(28): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#14 /composer/vendor/illuminate/pipeline/Pipeline.php(184): App\\Services\\Forge\\Pipeline\\CreateDatabase->__invoke()
#15 /composer/vendor/mehrancodes/laravel-harbor/app/Services/Forge/Pipeline/NginxTemplateSearchReplace.php(31): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#16 /composer/vendor/illuminate/pipeline/Pipeline.php(184): App\\Services\\Forge\\Pipeline\\NginxTemplateSearchReplace->__invoke()
#17 /composer/vendor/mehrancodes/laravel-harbor/app/Services/Forge/Pipeline/OrCreateNewSite.php(35): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#18 /composer/vendor/illuminate/pipeline/Pipeline.php(184): App\\Services\\Forge\\Pipeline\\OrCreateNewSite->__invoke()
#19 /composer/vendor/mehrancodes/laravel-harbor/app/Services/Forge/Pipeline/FindSite.php(32): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#20 /composer/vendor/illuminate/pipeline/Pipeline.php(184): App\\Services\\Forge\\Pipeline\\FindSite->__invoke()
#21 /composer/vendor/mehrancodes/laravel-harbor/app/Services/Forge/Pipeline/FindServer.php(34): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#22 /composer/vendor/illuminate/pipeline/Pipeline.php(184): App\\Services\\Forge\\Pipeline\\FindServer->__invoke()
#23 /composer/vendor/illuminate/pipeline/Pipeline.php(119): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#24 /composer/vendor/mehrancodes/laravel-harbor/app/Commands/ProvisionCommand.php(69): Illuminate\\Pipeline\\Pipeline->then()
#25 /composer/vendor/illuminate/container/BoundMethod.php(36): App\\Commands\\ProvisionCommand->handle()
#26 /composer/vendor/illuminate/container/Util.php(41): Illuminate\\Container\\BoundMethod::Illuminate\\Container\\{closure}()
#27 /composer/vendor/illuminate/container/BoundMethod.php(93): Illuminate\\Container\\Util::unwrapIfClosure()
#28 /composer/vendor/illuminate/container/BoundMethod.php(37): Illuminate\\Container\\BoundMethod::callBoundMethod()
#29 /composer/vendor/illuminate/container/Container.php(662): Illuminate\\Container\\BoundMethod::call()
#30 /composer/vendor/illuminate/console/Command.php(211): Illuminate\\Container\\Container->call()
#31 /composer/vendor/symfony/console/Command/Command.php(326): Illuminate\\Console\\Command->execute()
#32 /composer/vendor/illuminate/console/Command.php(181): Symfony\\Component\\Console\\Command\\Command->run()
#33 /composer/vendor/symfony/console/Application.php(1096): Illuminate\\Console\\Command->run()
#34 /composer/vendor/symfony/console/Application.php(324): Symfony\\Component\\Console\\Application->doRunCommand()
#35 /composer/vendor/symfony/console/Application.php(175): Symfony\\Component\\Console\\Application->doRun()
#36 /composer/vendor/laravel-zero/foundation/src/Illuminate/Foundation/Console/Kernel.php(201): Symfony\\Component\\Console\\Application->run()
#37 /composer/vendor/laravel-zero/framework/src/Kernel.php(97): Illuminate\\Foundation\\Console\\Kernel->handle()
#38 /composer/vendor/mehrancodes/laravel-harbor/harbor(37): LaravelZero\\Framework\\Kernel->handle()
#39 /composer/vendor/bin/harbor(119): include('...')
#40 {main}
"} 
   ErrorException 
  Undefined array key 1
  at /composer/vendor/mehrancodes/laravel-harbor/app/Actions/MergeEnvironmentVariables.php:53
     49▕ 
     50▕                 continue;
     51▕             }
     52▕ 
  ➜  53▕             [$key, $value] = explode('=', $variable, 2);
     54▕ 
     55▕             if (empty($key)) {
     56▕                 $this->warning("No key found for the assigned value \"$value\" inside your environment variables! Make sure to remove it.");
     57▕
```

Is there something incorrect in my gitlab-ci.yml file?

```
harbor-provision-staging-site:
  stage: deploy
  variables:
    FORGE_TOKEN: XXXXXXX
    FORGE_SERVER: XXXXXX
    FORGE_GIT_PROVIDER: gitlab
    FORGE_GIT_REPOSITORY: $CI_MERGE_REQUEST_PROJECT_PATH
    FORGE_GIT_BRANCH: $CI_MERGE_REQUEST_SOURCE_BRANCH_NAME
    FORGE_DOMAIN: sub-domain.domain.net
    FORGE_SUBDOMAIN_PATTERN: /TM-\d+/
    FORGE_ENV_KEYS: "STRIPE_KEY=XXXX; STRIPE_SECRET=XXXX;"
    FORGE_DEPLOY_SCRIPT: "cd $FORGE_SITE_PATH; git pull origin $FORGE_SITE_BRANCH; composer install; php artisan migrate --force;"
    FORGE_JOB_SCHEDULER: true
    FORGE_DB_CREATION_REQUIRED: true
    FORGE_DB_NAME: $CI_MERGE_REQUEST_SOURCE_BRANCH_NAME
  dependencies: []
  rules:
    - if: $CI_MERGE_REQUEST_PROJECT_PATH != '' # ensures the job only runs in merge request context
      when: manual
  image: kirschbaumdevelopment/laravel-test-runner:8.1
  script:
    # Install Harbor via Composer
    - composer global require mehrancodes/laravel-harbor -q
    # Start Provisioning
    - harbor provision
```

Thanks for any assitance

Me too, on GitHub. I set a variable for my env vars incase it was my yaml-fu letting me down.

```sh
##[debug]..=> 'ENV_VARS'
##[debug]=> '"APP_ENV=preview; VITE_APP_ENV=production; APP_DEBUG=false; INTERCOM_ACCESS_TOKEN=redacted; INTERCOM_API_VERSION=unstable; LOG_CHANNEL=single; BROADCAST_DRIVER=log; CACHE_DRIVER=redis; QUEUE_CONNECTION=redis; SESSION_DRIVER=redis; SESSION_LIFETIME=120; PENNANT_STORE=database; REDIS_CLIENT=phpredis; REDIS_HOST=127.0.0.1; REDIS_PORT=6379; MAIL_MAILER=postmark; POSTMARK_SECRET=redacted;"'
##[debug]Result: '"APP_ENV=preview; VITE_APP_ENV=production; APP_DEBUG=false; INTERCOM_ACCESS_TOKEN=redacted; INTERCOM_API_VERSION=unstable; LOG_CHANNEL=single; BROADCAST_DRIVER=log; CACHE_DRIVER=redis; QUEUE_CONNECTION=redis; SESSION_DRIVER=redis; SESSION_LIFETIME=120; PENNANT_STORE=database; REDIS_CLIENT=phpredis; REDIS_HOST=127.0.0.1; REDIS_PORT=6379; MAIL_MAILER=postmark; POSTMARK_SECRET=redacted;"'
##[debug]Evaluating condition for step: 'Start Provisioning'
##[debug]Evaluating: success()
##[debug]Evaluating success:
##[debug]=> true
##[debug]Result: true
##[debug]Starting: Start Provisioning
##[debug]Loading inputs
##[debug]Loading env
Run harbor provision
##[debug]sh -e /__w/_temp/dc012cfe-db4c-42ba-882e-b016c8a74334.sh
  INFO    Start finding the server.
  INFO    Finding the associated site.
  INFO    Processing update of environment variables.
[2024-04-12 11:31:25] development.ERROR: Undefined array key 1 {"exception":"[object] (ErrorException(code: 0): Undefined array key 1 at /composer/vendor/mehrancodes/laravel-harbor/app/Actions/MergeEnvironmentVariables.php:53)
[stacktrace]
#0 /composer/vendor/laravel-zero/foundation/src/Illuminate/Foundation/Bootstrap/HandleExceptions.php(255): Illuminate\\Foundation\\Bootstrap\\HandleExceptions->handleError()
#1 /composer/vendor/mehrancodes/laravel-harbor/app/Actions/MergeEnvironmentVariables.php(53): Illuminate\\Foundation\\Bootstrap\\HandleExceptions->Illuminate\\Foundation\\Bootstrap\\{closure}()
#2 /composer/vendor/mehrancodes/laravel-harbor/app/Actions/MergeEnvironmentVariables.php(31): App\\Actions\\MergeEnvironmentVariables->searchReplaceExistingVariables()
#3 /composer/vendor/lorisleiva/laravel-actions/src/Concerns/AsObject.php(24): App\\Actions\\MergeEnvironmentVariables->handle()
#4 /composer/vendor/mehrancodes/laravel-harbor/app/Services/Forge/Pipeline/UpdateEnvironmentVariables.php(47): App\\Actions\\MergeEnvironmentVariables::run()
#5 /composer/vendor/mehrancodes/laravel-harbor/app/Services/Forge/Pipeline/UpdateEnvironmentVariables.php(36): App\\Services\\Forge\\Pipeline\\UpdateEnvironmentVariables->getBothEnvsMerged()
#6 /composer/vendor/illuminate/pipeline/Pipeline.php(184): App\\Services\\Forge\\Pipeline\\UpdateEnvironmentVariables->__invoke()
#7 /composer/vendor/mehrancodes/laravel-harbor/app/Services/Forge/Pipeline/EnableQuickDeploy.php(27): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#8 /composer/vendor/illuminate/pipeline/Pipeline.php(184): App\\Services\\Forge\\Pipeline\\EnableQuickDeploy->__invoke()
#9 /composer/vendor/mehrancodes/laravel-harbor/app/Services/Forge/Pipeline/ObtainLetsEncryptCertification.php(28): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#10 /composer/vendor/illuminate/pipeline/Pipeline.php(184): App\\Services\\Forge\\Pipeline\\ObtainLetsEncryptCertification->__invoke()
#11 /composer/vendor/mehrancodes/laravel-harbor/app/Services/Forge/Pipeline/InstallGitRepository.php(27): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#12 /composer/vendor/illuminate/pipeline/Pipeline.php(184): App\\Services\\Forge\\Pipeline\\InstallGitRepository->__invoke()
#13 /composer/vendor/mehrancodes/laravel-harbor/app/Services/Forge/Pipeline/CreateDatabase.php(28): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#14 /composer/vendor/illuminate/pipeline/Pipeline.php(184): App\\Services\\Forge\\Pipeline\\CreateDatabase->__invoke()
#15 /composer/vendor/mehrancodes/laravel-harbor/app/Services/Forge/Pipeline/NginxTemplateSearchReplace.php(31): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#16 /composer/vendor/illuminate/pipeline/Pipeline.php(184): App\\Services\\Forge\\Pipeline\\NginxTemplateSearchReplace->__invoke()
#17 /composer/vendor/mehrancodes/laravel-harbor/app/Services/Forge/Pipeline/OrCreateNewSite.php(35): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#18 /composer/vendor/illuminate/pipeline/Pipeline.php(184): App\\Services\\Forge\\Pipeline\\OrCreateNewSite->__invoke()
#19 /composer/vendor/mehrancodes/laravel-harbor/app/Services/Forge/Pipeline/FindSite.php(32): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#20 /composer/vendor/illuminate/pipeline/Pipeline.php(184): App\\Services\\Forge\\Pipeline\\FindSite->__invoke()
#21 /composer/vendor/mehrancodes/laravel-harbor/app/Services/Forge/Pipeline/FindServer.php(34): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#22 /composer/vendor/illuminate/pipeline/Pipeline.php(184): App\\Services\\Forge\\Pipeline\\FindServer->__invoke()
#23 /composer/vendor/illuminate/pipeline/Pipeline.php(119): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#24 /composer/vendor/mehrancodes/laravel-harbor/app/Commands/ProvisionCommand.php(66): Illuminate\\Pipeline\\Pipeline->then()
#25 /composer/vendor/illuminate/container/BoundMethod.php(36): App\\Commands\\ProvisionCommand->handle()
#26 /composer/vendor/illuminate/container/Util.php(41): Illuminate\\Container\\BoundMethod::Illuminate\\Container\\{closure}()
#27 /composer/vendor/illuminate/container/BoundMethod.php(93): Illuminate\\Container\\Util::unwrapIfClosure()
#28 /composer/vendor/illuminate/container/BoundMethod.php(35): Illuminate\\Container\\BoundMethod::callBoundMethod()
#29 /composer/vendor/illuminate/container/Container.php(662): Illuminate\\Container\\BoundMethod::call()
#30 /composer/vendor/illuminate/console/Command.php(211): Illuminate\\Container\\Container->call()
#31 /composer/vendor/symfony/console/Command/Command.php(326): Illuminate\\Console\\Command->execute()
#32 /composer/vendor/illuminate/console/Command.php(180): Symfony\\Component\\Console\\Command\\Command->run()
#33 /composer/vendor/symfony/console/Application.php(1096): Illuminate\\Console\\Command->run()
#34 /composer/vendor/symfony/console/Application.php(324): Symfony\\Component\\Console\\Application->doRunCommand()
#35 /composer/vendor/symfony/console/Application.php(175): Symfony\\Component\\Console\\Application->doRun()
#36 /composer/vendor/laravel-zero/foundation/src/Illuminate/Foundation/Console/Kernel.php(201): Symfony\\Component\\Console\\Application->run()
#37 /composer/vendor/laravel-zero/framework/src/Kernel.php(97): Illuminate\\Foundation\\Console\\Kernel->handle()
#38 /composer/vendor/mehrancodes/laravel-harbor/harbor(35): LaravelZero\\Framework\\Kernel->handle()
#39 /composer/vendor/bin/harbor(119): include('...')
#40 {main}
"} 
   ErrorException 
  Undefined array key 1
  at /composer/vendor/mehrancodes/laravel-harbor/app/Actions/MergeEnvironmentVariables.php:53
     49▕ 
     50▕                 continue;
     51▕             }
     52▕ 
  ➜  53▕             [$key, $value] = explode('=', $variable, 2);
     54▕ 
     55▕             if (empty($key)) {
     56▕                 $this->warning("No key found for the assigned value \"$value\" inside your environment variables! Make sure to remove it.");
     57▕
Error: Process completed with exit code 1.
##[debug]Finishing: Start Provisioning
```

> @neilgilmour It should be Forge API throwing an excepting, but not clear where it does happen. Could you sebd the full stack trace here or in a new issue?

Sure! zip file attached for full logs, and here's relevant file;


```sh
2024-04-12T13:22:44.0370118Z ##[debug]Starting: harbor-provision
2024-04-12T13:22:44.0397435Z ##[debug]Cleaning runner temp folder: /home/runner/work/_temp
2024-04-12T13:22:44.0678702Z ##[debug]Starting: Set up job
2024-04-12T13:22:44.0679713Z Current runner version: '2.315.0'
2024-04-12T13:22:44.0709668Z ##[group]Operating System
2024-04-12T13:22:44.0710475Z Ubuntu
2024-04-12T13:22:44.0710879Z 22.04.4
2024-04-12T13:22:44.0711241Z LTS
2024-04-12T13:22:44.0711676Z ##[endgroup]
2024-04-12T13:22:44.0712092Z ##[group]Runner Image
2024-04-12T13:22:44.0712572Z Image: ubuntu-22.04
2024-04-12T13:22:44.0713112Z Version: 20240407.1.0
2024-04-12T13:22:44.0714122Z Included Software: https://github.com/actions/runner-images/blob/ubuntu22/20240407.1/images/ubuntu/Ubuntu2204-Readme.md
2024-04-12T13:22:44.0715642Z Image Release: https://github.com/actions/runner-images/releases/tag/ubuntu22%2F20240407.1
2024-04-12T13:22:44.0716659Z ##[endgroup]
2024-04-12T13:22:44.0717096Z ##[group]Runner Image Provisioner
2024-04-12T13:22:44.0717587Z 2.0.361.1
2024-04-12T13:22:44.0718277Z ##[endgroup]
2024-04-12T13:22:44.0720781Z ##[group]GITHUB_TOKEN Permissions
2024-04-12T13:22:44.0722871Z Actions: write
2024-04-12T13:22:44.0723315Z Attestations: write
2024-04-12T13:22:44.0723878Z Checks: write
2024-04-12T13:22:44.0724222Z Contents: write
2024-04-12T13:22:44.0724624Z Deployments: write
2024-04-12T13:22:44.0725105Z Discussions: write
2024-04-12T13:22:44.0725478Z Issues: write
2024-04-12T13:22:44.0725854Z Metadata: read
2024-04-12T13:22:44.0726289Z Packages: write
2024-04-12T13:22:44.0726700Z Pages: write
2024-04-12T13:22:44.0727042Z PullRequests: writ
[logs_22700753169.zip](https://github.com/mehrancodes/laravel-harbor/files/14999533/logs_22700753169.zip)
e
2024-04-12T13:22:44.0727530Z RepositoryProjects: write
2024-04-12T13:22:44.0728574Z SecurityEvents: write
2024-04-12T13:22:44.0729126Z Statuses: write
2024-04-12T13:22:44.0729690Z ##[endgroup]
2024-04-12T13:22:44.0733270Z Secret source: Actions
2024-04-12T13:22:44.0734270Z ##[debug]Primary repository: motional-io/motional2
2024-04-12T13:22:44.0735005Z Prepare workflow directory
2024-04-12T13:22:44.0813665Z ##[debug]Creating pipeline directory: '/home/runner/work/motional2'
2024-04-12T13:22:44.0817365Z ##[debug]Creating workspace directory: '/home/runner/work/motional2/motional2'
2024-04-12T13:22:44.0819152Z ##[debug]Update context data
2024-04-12T13:22:44.0823201Z ##[debug]Evaluating job-level environment variables
2024-04-12T13:22:44.1357737Z ##[debug]Evaluating job container
2024-04-12T13:22:44.1612497Z ##[debug]Evaluating job service containers
2024-04-12T13:22:44.1615236Z ##[debug]Evaluating job defaults
2024-04-12T13:22:44.1693728Z Prepare all required actions
2024-04-12T13:22:44.2659669Z ##[debug]Set step '__run' display name to: 'Install Harbor via Composer'
2024-04-12T13:22:44.2662582Z ##[debug]Set step '__run_2' display name to: 'Start Provisioning'
2024-04-12T13:22:44.2664183Z Complete job name: harbor-provision
2024-04-12T13:22:44.2685370Z ##[debug]Collect running processes for tracking orphan processes.
2024-04-12T13:22:44.2919562Z ##[debug]Finishing: Set up job
2024-04-12T13:22:44.3154816Z ##[debug]Evaluating condition for step: 'Initialize containers'
2024-04-12T13:22:44.3299568Z ##[debug]Evaluating: success()
2024-04-12T13:22:44.3305961Z ##[debug]Evaluating success:
2024-04-12T13:22:44.3334174Z ##[debug]=> true
2024-04-12T13:22:44.3342516Z ##[debug]Result: true
2024-04-12T13:22:44.3379776Z ##[debug]Starting: Initialize containers
2024-04-12T13:22:44.3456284Z ##[debug]Register post job cleanup for stopping/deleting containers.
2024-04-12T13:22:44.3483826Z ##[group]Checking docker version
2024-04-12T13:22:44.3500186Z ##[command]/usr/bin/docker version --format '{{.Server.APIVersion}}'
2024-04-12T13:22:44.4557637Z '1.43'
2024-04-12T13:22:44.4580085Z Docker daemon API version: '1.43'
2024-04-12T13:22:44.4581153Z ##[command]/usr/bin/docker version --format '{{.Client.APIVersion}}'
2024-04-12T13:22:44.4785735Z '1.43'
2024-04-12T13:22:44.4799909Z Docker client API version: '1.43'
2024-04-12T13:22:44.4806012Z ##[endgroup]
2024-04-12T13:22:44.4810684Z ##[group]Clean up resources from previous jobs
2024-04-12T13:22:44.4818125Z ##[command]/usr/bin/docker ps --all --quiet --no-trunc --filter "label=c2eb9a"
2024-04-12T13:22:44.5129348Z ##[command]/usr/bin/docker network prune --force --filter "label=c2eb9a"
2024-04-12T13:22:44.5340568Z ##[endgroup]
2024-04-12T13:22:44.5341385Z ##[group]Create local container network
2024-04-12T13:22:44.5356277Z ##[command]/usr/bin/docker network create --label c2eb9a github_network_da9f8e9124b149dc95ad04d22529e2d8
2024-04-12T13:22:44.6363000Z fc9af473165839fcb555dfde7497a6ac1feb99f223e5b41be28bccf5b78f81d7
2024-04-12T13:22:44.6383282Z ##[endgroup]
2024-04-12T13:22:44.6467291Z ##[group]Starting job container
2024-04-12T13:22:44.6522630Z ##[command]/usr/bin/docker pull kirschbaumdevelopment/laravel-test-runner:8.3
2024-04-12T13:22:45.8006482Z 8.3: Pulling from kirschbaumdevelopment/laravel-test-runner
2024-04-12T13:22:46.0385828Z 57c139bbda7e: Pulling fs layer
2024-04-12T13:22:46.0386596Z ed1cfd3c5a16: Pulling fs layer
2024-04-12T13:22:46.0387070Z 3d734cee07ec: Pulling fs layer
2024-04-12T13:22:46.0387438Z 4a14e0bc620b: Pulling fs layer
2024-04-12T13:22:46.0387902Z 4f4fb700ef54: Pulling fs layer
2024-04-12T13:22:46.0388297Z 65099eff495c: Pulling fs layer
2024-04-12T13:22:46.0388677Z 52afd21137ea: Pulling fs layer
2024-04-12T13:22:46.0389112Z c2bb03afe941: Pulling fs layer
2024-04-12T13:22:46.0389492Z 62d54b6bf8a9: Pulling fs layer
2024-04-12T13:22:46.0389869Z c25fb8eb1e34: Pulling fs layer
2024-04-12T13:22:46.0390342Z 1415422de28d: Pulling fs layer
2024-04-12T13:22:46.0390690Z e1d9e72e0946: Pulling fs layer
2024-04-12T13:22:46.0391076Z 968b591e3cc8: Pulling fs layer
2024-04-12T13:22:46.0391580Z 4a14e0bc620b: Waiting
2024-04-12T13:22:46.0391902Z 4f4fb700ef54: Waiting
2024-04-12T13:22:46.0392238Z 65099eff495c: Waiting
2024-04-12T13:22:46.0392660Z 52afd21137ea: Waiting
2024-04-12T13:22:46.0392995Z c2bb03afe941: Waiting
2024-04-12T13:22:46.0393304Z 62d54b6bf8a9: Waiting
2024-04-12T13:22:46.0393691Z c25fb8eb1e34: Waiting
2024-04-12T13:22:46.0394022Z 1415422de28d: Waiting
2024-04-12T13:22:46.0394346Z e1d9e72e0946: Waiting
2024-04-12T13:22:46.0394752Z 968b591e3cc8: Waiting
2024-04-12T13:22:46.3408318Z 3d734cee07ec: Verifying Checksum
2024-04-12T13:22:46.3409485Z 3d734cee07ec: Download complete
2024-04-12T13:22:46.6213398Z 57c139bbda7e: Verifying Checksum
2024-04-12T13:22:46.6214519Z 57c139bbda7e: Download complete
2024-04-12T13:22:46.7465193Z 4a14e0bc620b: Verifying Checksum
2024-04-12T13:22:46.7472281Z 4a14e0bc620b: Download complete
2024-04-12T13:22:46.9355435Z 4f4fb700ef54: Verifying Checksum
2024-04-12T13:22:46.9356450Z 4f4fb700ef54: Download complete
2024-04-12T13:22:46.9839356Z ed1cfd3c5a16: Verifying Checksum
2024-04-12T13:22:46.9842614Z ed1cfd3c5a16: Download complete
2024-04-12T13:22:47.2221172Z 65099eff495c: Verifying Checksum
2024-04-12T13:22:47.2222039Z 65099eff495c: Download complete
2024-04-12T13:22:47.2522210Z 52afd21137ea: Verifying Checksum
2024-04-12T13:22:47.2523084Z 52afd21137ea: Download complete
2024-04-12T13:22:47.4143865Z c2bb03afe941: Verifying Checksum
2024-04-12T13:22:47.4147350Z c2bb03afe941: Download complete
2024-04-12T13:22:47.5606404Z c25fb8eb1e34: Verifying Checksum
2024-04-12T13:22:47.5621684Z c25fb8eb1e34: Download complete
2024-04-12T13:22:47.7202965Z 1415422de28d: Verifying Checksum
2024-04-12T13:22:47.7204070Z 1415422de28d: Download complete
2024-04-12T13:22:47.7396699Z 62d54b6bf8a9: Verifying Checksum
2024-04-12T13:22:47.7397489Z 62d54b6bf8a9: Download complete
2024-04-12T13:22:48.0835077Z 968b591e3cc8: Verifying Checksum
2024-04-12T13:22:48.0835856Z 968b591e3cc8: Download complete
2024-04-12T13:22:48.2616460Z e1d9e72e0946: Verifying Checksum
2024-04-12T13:22:48.2617358Z e1d9e72e0946: Download complete
2024-04-12T13:22:48.4889649Z 57c139bbda7e: Pull complete
2024-04-12T13:22:52.8933855Z ed1cfd3c5a16: Pull complete
2024-04-12T13:22:52.9344398Z 3d734cee07ec: Pull complete
2024-04-12T13:22:52.9765982Z 4a14e0bc620b: Pull complete
2024-04-12T13:22:52.9891570Z 4f4fb700ef54: Pull complete
2024-04-12T13:22:54.3345513Z 65099eff495c: Pull complete
2024-04-12T13:22:54.5445417Z 52afd21137ea: Pull complete
2024-04-12T13:22:55.0962842Z c2bb03afe941: Pull complete
2024-04-12T13:22:56.3416295Z 62d54b6bf8a9: Pull complete
2024-04-12T13:22:56.3718636Z c25fb8eb1e34: Pull complete
2024-04-12T13:22:56.3895776Z 1415422de28d: Pull complete
2024-04-12T13:22:59.4894516Z e1d9e72e0946: Pull complete
2024-04-12T13:22:59.5923957Z 968b591e3cc8: Pull complete
2024-04-12T13:22:59.5971493Z Digest: sha256:c0d47c71ed3e8570fb24cf142ff8bc2cba6535474a9d6d2757a577afb4361a72
2024-04-12T13:22:59.5983378Z Status: Downloaded newer image for kirschbaumdevelopment/laravel-test-runner:8.3
2024-04-12T13:22:59.5998186Z docker.io/kirschbaumdevelopment/laravel-test-runner:8.3
2024-04-12T13:22:59.6144119Z ##[command]/usr/bin/docker create --name a4b1c5e61e744335aaf2387a9e8d1457_kirschbaumdevelopmentlaraveltestrunner83_9f0f06 --label c2eb9a --workdir /__w/motional2/motional2 --network github_network_da9f8e9124b149dc95ad04d22529e2d8  -e "HOME=/github/home" -e GITHUB_ACTIONS=true -e CI=true -v "/var/run/docker.sock":"/var/run/docker.sock" -v "/home/runner/work":"/__w" -v "/home/runner/runners/2.315.0/externals":"/__e":ro -v "/home/runner/work/_temp":"/__w/_temp" -v "/home/runner/work/_actions":"/__w/_actions" -v "/opt/hostedtoolcache":"/__t" -v "/home/runner/work/_temp/_github_home":"/github/home" -v "/home/runner/work/_temp/_github_workflow":"/github/workflow" --entrypoint "tail" kirschbaumdevelopment/laravel-test-runner:8.3 "-f" "/dev/null"
2024-04-12T13:22:59.6479162Z c5b0276a75a8bb2849a46fccf3b73b5df39f8da1d05d54e2917015afa5af7ed3
2024-04-12T13:22:59.6507752Z ##[command]/usr/bin/docker start c5b0276a75a8bb2849a46fccf3b73b5df39f8da1d05d54e2917015afa5af7ed3
2024-04-12T13:22:59.9355076Z c5b0276a75a8bb2849a46fccf3b73b5df39f8da1d05d54e2917015afa5af7ed3
2024-04-12T13:22:59.9375036Z ##[command]/usr/bin/docker ps --all --filter id=c5b0276a75a8bb2849a46fccf3b73b5df39f8da1d05d54e2917015afa5af7ed3 --filter status=running --no-trunc --format "{{.ID}} {{.Status}}"
2024-04-12T13:22:59.9510856Z c5b0276a75a8bb2849a46fccf3b73b5df39f8da1d05d54e2917015afa5af7ed3 Up Less than a second
2024-04-12T13:22:59.9532691Z ##[command]/usr/bin/docker inspect --format "{{range .Config.Env}}{{println .}}{{end}}" c5b0276a75a8bb2849a46fccf3b73b5df39f8da1d05d54e2917015afa5af7ed3
2024-04-12T13:22:59.9665291Z HOME=/github/home
2024-04-12T13:22:59.9678915Z GITHUB_ACTIONS=true
2024-04-12T13:22:59.9679612Z CI=true
2024-04-12T13:22:59.9680739Z PATH=./vendor/bin:/composer/vendor/bin:/root/.yarn/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
2024-04-12T13:22:59.9681989Z PHP_VERSION=8.3
2024-04-12T13:22:59.9682569Z GOSS_VERSION=0.4.4
2024-04-12T13:22:59.9683084Z NODE_MAJOR=20
2024-04-12T13:22:59.9683614Z COMPOSER_HOME=/composer
2024-04-12T13:22:59.9684248Z COMPOSER_ALLOW_SUPERUSER=1
2024-04-12T13:22:59.9697910Z ##[endgroup]
2024-04-12T13:22:59.9724412Z ##[group]Waiting for all services to be ready
2024-04-12T13:22:59.9726320Z ##[endgroup]
2024-04-12T13:22:59.9735496Z ##[debug]Finishing: Initialize containers
2024-04-12T13:22:59.9752612Z ##[debug]Evaluating condition for step: 'Install Harbor via Composer'
2024-04-12T13:22:59.9757771Z ##[debug]Evaluating: success()
2024-04-12T13:22:59.9758522Z ##[debug]Evaluating success:
2024-04-12T13:22:59.9759265Z ##[debug]=> true
2024-04-12T13:22:59.9760024Z ##[debug]Result: true
2024-04-12T13:22:59.9761138Z ##[debug]Starting: Install Harbor via Composer
2024-04-12T13:23:00.0060880Z ##[debug]Loading inputs
2024-04-12T13:23:00.0084167Z ##[debug]Loading env
2024-04-12T13:23:00.0182672Z ##[group]Run composer global require mehrancodes/laravel-harbor -q
2024-04-12T13:23:00.0183438Z [36;1mcomposer global require mehrancodes/laravel-harbor -q[0m
2024-04-12T13:23:00.0186927Z shell: sh -e {0}
2024-04-12T13:23:00.0187291Z ##[endgroup]
2024-04-12T13:23:00.0364233Z ##[debug]sh -e /__w/_temp/60fa9929-1747-4d4c-8fd4-430925f9f072.sh
2024-04-12T13:23:07.0781900Z ##[debug]Finishing: Install Harbor via Composer
2024-04-12T13:23:07.0807289Z ##[debug]Evaluating: secrets.FORGE_TOKEN
2024-04-12T13:23:07.0809091Z ##[debug]Evaluating Index:
2024-04-12T13:23:07.0811413Z ##[debug]..Evaluating secrets:
2024-04-12T13:23:07.0812741Z ##[debug]..=> Object
2024-04-12T13:23:07.0825165Z ##[debug]..Evaluating String:
2024-04-12T13:23:07.0826137Z ##[debug]..=> 'FORGE_TOKEN'
2024-04-12T13:23:07.0836021Z ##[debug]=> '***'
2024-04-12T13:23:07.0843399Z ##[debug]Result: '***'
2024-04-12T13:23:07.0844710Z ##[debug]Evaluating: github.repository
2024-04-12T13:23:07.0845202Z ##[debug]Evaluating Index:
2024-04-12T13:23:07.0845731Z ##[debug]..Evaluating github:
2024-04-12T13:23:07.0846121Z ##[debug]..=> Object
2024-04-12T13:23:07.0846690Z ##[debug]..Evaluating String:
2024-04-12T13:23:07.0847200Z ##[debug]..=> 'repository'
2024-04-12T13:23:07.0848089Z ##[debug]=> 'motional-io/motional2'
2024-04-12T13:23:07.0848810Z ##[debug]Result: 'motional-io/motional2'
2024-04-12T13:23:07.0849573Z ##[debug]Evaluating: github.head_ref
2024-04-12T13:23:07.0850271Z ##[debug]Evaluating Index:
2024-04-12T13:23:07.0850648Z ##[debug]..Evaluating github:
2024-04-12T13:23:07.0851065Z ##[debug]..=> Object
2024-04-12T13:23:07.0851509Z ##[debug]..Evaluating String:
2024-04-12T13:23:07.0851899Z ##[debug]..=> 'head_ref'
2024-04-12T13:23:07.0852332Z ##[debug]=> 'harbor'
2024-04-12T13:23:07.0852776Z ##[debug]Result: 'harbor'
2024-04-12T13:23:07.0853497Z ##[debug]Evaluating: secrets.GITHUB_TOKEN
2024-04-12T13:23:07.0853956Z ##[debug]Evaluating Index:
2024-04-12T13:23:07.0854400Z ##[debug]..Evaluating secrets:
2024-04-12T13:23:07.0854813Z ##[debug]..=> Object
2024-04-12T13:23:07.0855170Z ##[debug]..Evaluating String:
2024-04-12T13:23:07.0855646Z ##[debug]..=> 'GITHUB_TOKEN'
2024-04-12T13:23:07.0856236Z ##[debug]=> '***'
2024-04-12T13:23:07.0856804Z ##[debug]Result: '***'
2024-04-12T13:23:07.0857494Z ##[debug]Evaluating: github.event.number
2024-04-12T13:23:07.0857961Z ##[debug]Evaluating Index:
2024-04-12T13:23:07.0858458Z ##[debug]..Evaluating Index:
2024-04-12T13:23:07.0858860Z ##[debug]....Evaluating github:
2024-04-12T13:23:07.0859248Z ##[debug]....=> Object
2024-04-12T13:23:07.0859711Z ##[debug]....Evaluating String:
2024-04-12T13:23:07.0860126Z ##[debug]....=> 'event'
2024-04-12T13:23:07.0860510Z ##[debug]..=> Object
2024-04-12T13:23:07.0860971Z ##[debug]..Evaluating String:
2024-04-12T13:23:07.0861318Z ##[debug]..=> 'number'
2024-04-12T13:23:07.0862588Z ##[debug]=> 418
2024-04-12T13:23:07.0863144Z ##[debug]Result: 418
2024-04-12T13:23:07.0866017Z ##[debug]Evaluating: vars.DEPLOY_SCRIPT
2024-04-12T13:23:07.0866612Z ##[debug]Evaluating Index:
2024-04-12T13:23:07.0867023Z ##[debug]..Evaluating vars:
2024-04-12T13:23:07.0867425Z ##[debug]..=> Object
2024-04-12T13:23:07.0867886Z ##[debug]..Evaluating String:
2024-04-12T13:23:07.0868302Z ##[debug]..=> 'DEPLOY_SCRIPT'
2024-04-12T13:23:07.0868894Z ##[debug]=> '#! /usr/bin/env bash
2024-04-12T13:23:07.0869360Z ##[debug]
2024-04-12T13:23:07.0869663Z ##[debug]set -e
2024-04-12T13:23:07.0870059Z ##[debug]
2024-04-12T13:23:07.0870348Z ##[debug]cd $FORGE_SITE_PATH
2024-04-12T13:23:07.0870756Z ##[debug]echo "pwd is $(pwd)"
2024-04-12T13:23:07.0871191Z ##[debug]
2024-04-12T13:23:07.0871537Z ##[debug]git pull origin $FORGE_SITE_BRANCH
2024-04-12T13:23:07.0871940Z ##[debug]
2024-04-12T13:23:07.0872390Z ##[debug]echo "🛠️ Building Stuff"
2024-04-12T13:23:07.0873041Z ##[debug]$FORGE_COMPOSER install --no-interaction --no-dev --prefer-dist
2024-04-12T13:23:07.0873570Z ##[debug]npm ci
2024-04-12T13:23:07.0873963Z ##[debug]npm run build
2024-04-12T13:23:07.0874306Z ##[debug]
2024-04-12T13:23:07.0874722Z ##[debug]echo "🗑️ Removing Node Modules"
2024-04-12T13:23:07.0875235Z ##[debug]rm -rf ./node_modules
2024-04-12T13:23:07.0875586Z ##[debug]
2024-04-12T13:23:07.0875955Z ##[debug]echo "Quick Deploy is: $FORGE_QUICK_DEPLOY"
2024-04-12T13:23:07.0876546Z ##[debug]if [[ $FORGE_QUICK_DEPLOY -eq 0 ]]; then
2024-04-12T13:23:07.0876994Z ##[debug]
2024-04-12T13:23:07.0877319Z ##[debug]  echo "💽 adding dummy data to the DB."
2024-04-12T13:23:07.0878360Z ##[debug]  php artisan db:anonymous
2024-04-12T13:23:07.0878809Z ##[debug]  php artisan key:generate
2024-04-12T13:23:07.0879201Z ##[debug]fi
2024-04-12T13:23:07.0879743Z ##[debug]
2024-04-12T13:23:07.0880065Z ##[debug]php artisan migrate --force
2024-04-12T13:23:07.0880516Z ##[debug]php artisan horizon:terminate
2024-04-12T13:23:07.0880945Z ##[debug]
2024-04-12T13:23:07.0881284Z ##[debug]echo "💾 Caching and Optimizing."
2024-04-12T13:23:07.0881720Z ##[debug]
2024-04-12T13:23:07.0882102Z ##[debug]php artisan route:cache
2024-04-12T13:23:07.0882470Z ##[debug]php artisan event:cache
2024-04-12T13:23:07.0882887Z ##[debug]php artisan view:cache
2024-04-12T13:23:07.0883360Z ##[debug]php artisan config:cache
2024-04-12T13:23:07.0883771Z ##[debug]php artisan telescope:prune
2024-04-12T13:23:07.0884161Z ##[debug]
2024-04-12T13:23:07.0884574Z ##[debug]php artisan down --secret=bongo
2024-04-12T13:23:07.0885110Z ##[debug]echo "💯 Application deployed in Maintenance mode!"
2024-04-12T13:23:07.0885659Z ##[debug]'
2024-04-12T13:23:07.0886118Z ##[debug]Result: '#! /usr/bin/env bash
2024-04-12T13:23:07.0886493Z ##[debug]
2024-04-12T13:23:07.0886859Z ##[debug]set -e
2024-04-12T13:23:07.0887198Z ##[debug]
2024-04-12T13:23:07.0887474Z ##[debug]cd $FORGE_SITE_PATH
2024-04-12T13:23:07.0888264Z ##[debug]echo "pwd is $(pwd)"
2024-04-12T13:23:07.0888654Z ##[debug]
2024-04-12T13:23:07.0888989Z ##[debug]git pull origin $FORGE_SITE_BRANCH
2024-04-12T13:23:07.0889451Z ##[debug]
2024-04-12T13:23:07.0889813Z ##[debug]echo "🛠️ Building Stuff"
2024-04-12T13:23:07.0890381Z ##[debug]$FORGE_COMPOSER install --no-interaction --no-dev --prefer-dist
2024-04-12T13:23:07.0890989Z ##[debug]npm ci
2024-04-12T13:23:07.0891294Z ##[debug]npm run build
2024-04-12T13:23:07.0891630Z ##[debug]
2024-04-12T13:23:07.0892043Z ##[debug]echo "🗑️ Removing Node Modules"
2024-04-12T13:23:07.0892436Z ##[debug]rm -rf ./node_modules
2024-04-12T13:23:07.0892797Z ##[debug]
2024-04-12T13:23:07.0893240Z ##[debug]echo "Quick Deploy is: $FORGE_QUICK_DEPLOY"
2024-04-12T13:23:07.0893781Z ##[debug]if [[ $FORGE_QUICK_DEPLOY -eq 0 ]]; then
2024-04-12T13:23:07.0894181Z ##[debug]
2024-04-12T13:23:07.0894644Z ##[debug]  echo "💽 adding dummy data to the DB."
2024-04-12T13:23:07.0895138Z ##[debug]  php artisan db:anonymous
2024-04-12T13:23:07.0895615Z ##[debug]  php artisan key:generate
2024-04-12T13:23:07.0896009Z ##[debug]fi
2024-04-12T13:23:07.0896304Z ##[debug]
2024-04-12T13:23:07.0896718Z ##[debug]php artisan migrate --force
2024-04-12T13:23:07.0897145Z ##[debug]php artisan horizon:terminate
2024-04-12T13:23:07.0897510Z ##[debug]
2024-04-12T13:23:07.0897920Z ##[debug]echo "💾 Caching and Optimizing."
2024-04-12T13:23:07.0898346Z ##[debug]
2024-04-12T13:23:07.0898629Z ##[debug]php artisan route:cache
2024-04-12T13:23:07.0899116Z ##[debug]php artisan event:cache
2024-04-12T13:23:07.0899517Z ##[debug]php artisan view:cache
2024-04-12T13:23:07.0899943Z ##[debug]php artisan config:cache
2024-04-12T13:23:07.0900395Z ##[debug]php artisan telescope:prune
2024-04-12T13:23:07.0900796Z ##[debug]
2024-04-12T13:23:07.0901146Z ##[debug]php artisan down --secret=bongo
2024-04-12T13:23:07.0901728Z ##[debug]echo "💯 Application deployed in Maintenance mode!"
2024-04-12T13:23:07.0902187Z ##[debug]'
2024-04-12T13:23:07.0916608Z ##[debug]Evaluating condition for step: 'Start Provisioning'
2024-04-12T13:23:07.0918746Z ##[debug]Evaluating: success()
2024-04-12T13:23:07.0919501Z ##[debug]Evaluating success:
2024-04-12T13:23:07.0920146Z ##[debug]=> true
2024-04-12T13:23:07.0920906Z ##[debug]Result: true
2024-04-12T13:23:07.0921707Z ##[debug]Starting: Start Provisioning
2024-04-12T13:23:07.0939738Z ##[debug]Loading inputs
2024-04-12T13:23:07.0940820Z ##[debug]Loading env
2024-04-12T13:23:07.0948272Z ##[group]Run harbor provision
2024-04-12T13:23:07.0948694Z [36;1mharbor provision[0m
2024-04-12T13:23:07.0949176Z shell: sh -e {0}
2024-04-12T13:23:07.0949530Z env:
2024-04-12T13:23:07.0955484Z   FORGE_TOKEN: ***
2024-04-12T13:23:07.0955942Z   FORGE_SERVER: 769642
2024-04-12T13:23:07.0956577Z   FORGE_GIT_REPOSITORY: motional-io/motional2
2024-04-12T13:23:07.0957018Z   FORGE_GIT_BRANCH: harbor
2024-04-12T13:23:07.0957492Z   FORGE_DOMAIN: motional.dev
2024-04-12T13:23:07.0958047Z   GIT_COMMENT_ENABLED: true
2024-04-12T13:23:07.0958621Z   GIT_TOKEN: ***
2024-04-12T13:23:07.0958920Z   GIT_ISSUE_NUMBER: 418
2024-04-12T13:23:07.0962283Z   FORGE_DEPLOY_SCRIPT: #! /usr/bin/env bash

set -e

cd $FORGE_SITE_PATH
echo "pwd is $(pwd)"

git pull origin $FORGE_SITE_BRANCH

echo "🛠️ Building Stuff"
$FORGE_COMPOSER install --no-interaction --no-dev --prefer-dist
npm ci
npm run build

echo "🗑️ Removing Node Modules"
rm -rf ./node_modules

echo "Quick Deploy is: $FORGE_QUICK_DEPLOY"
if [[ $FORGE_QUICK_DEPLOY -eq 0 ]]; then

  echo "💽 adding dummy data to the DB."
  php artisan db:anonymous
  php artisan key:generate
fi

php artisan migrate --force
php artisan horizon:terminate

echo "💾 Caching and Optimizing."

php artisan route:cache
php artisan event:cache
php artisan view:cache
php artisan config:cache
php artisan telescope:prune

php artisan down --secret=bongo
echo "💯 Application deployed in Maintenance mode!"

2024-04-12T13:23:07.0965639Z   FORGE_NGINX_TEMPLATE: 11759
2024-04-12T13:23:07.0966003Z   FORGE_PHP_VERSION: php83
2024-04-12T13:23:07.0966402Z   FORGE_JOB_SCHEDULER: true
2024-04-12T13:23:07.0966864Z   FORGE_DB_CREATION_REQUIRED: true
2024-04-12T13:23:07.0967274Z   FORGE_SSL_REQUIRED: true
2024-04-12T13:23:07.0967620Z ##[endgroup]
2024-04-12T13:23:07.1029038Z ##[debug]sh -e /__w/_temp/b961b25b-d0cb-449a-8dc9-e1482a54fa5e.sh
2024-04-12T13:23:07.2972582Z   INFO    Start finding the server.
2024-04-12T13:23:07.9066196Z   INFO    Finding the associated site.
2024-04-12T13:23:08.6451061Z   INFO    Updating deployment script.
2024-04-12T13:23:09.4176770Z [2024-04-12 13:23:09] development.ERROR: {
2024-04-12T13:23:09.4178262Z     "message": "Server Error"
2024-04-12T13:23:09.4181422Z } {"exception":"[object] (Exception(code: 0): {
2024-04-12T13:23:09.4283500Z     \"message\": \"Server Error\"
2024-04-12T13:23:09.4284850Z } at /composer/vendor/laravel/forge-sdk/src/MakesHttpRequests.php:125)
2024-04-12T13:23:09.4285821Z [stacktrace]
2024-04-12T13:23:09.4287115Z #0 /composer/vendor/laravel/forge-sdk/src/MakesHttpRequests.php(83): Laravel\\Forge\\Forge->handleRequestError()
2024-04-12T13:23:09.4324759Z #1 /composer/vendor/laravel/forge-sdk/src/MakesHttpRequests.php(47): Laravel\\Forge\\Forge->request()
2024-04-12T13:23:09.4327069Z #2 /composer/vendor/laravel/forge-sdk/src/Actions/ManagesSites.php(246): Laravel\\Forge\\Forge->put()
2024-04-12T13:23:09.4329814Z #3 /composer/vendor/mehrancodes/laravel-harbor/app/Services/Forge/Pipeline/UpdateDeployScript.php(32): Laravel\\Forge\\Forge->updateSiteDeploymentScript()
2024-04-12T13:23:09.4332540Z #4 /composer/vendor/illuminate/pipeline/Pipeline.php(184): App\\Services\\Forge\\Pipeline\\UpdateDeployScript->__invoke()
2024-04-12T13:23:09.4335406Z #5 /composer/vendor/mehrancodes/laravel-harbor/app/Services/Forge/Pipeline/UpdateEnvironmentVariables.php(40): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
2024-04-12T13:23:09.4389180Z #6 /composer/vendor/illuminate/pipeline/Pipeline.php(184): App\\Services\\Forge\\Pipeline\\UpdateEnvironmentVariables->__invoke()
2024-04-12T13:23:09.4392064Z #7 /composer/vendor/mehrancodes/laravel-harbor/app/Services/Forge/Pipeline/EnableQuickDeploy.php(27): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
2024-04-12T13:23:09.4393684Z #8 /composer/vendor/illuminate/pipeline/Pipeline.php(184): App\\Services\\Forge\\Pipeline\\EnableQuickDeploy->__invoke()
2024-04-12T13:23:09.4395367Z #9 /composer/vendor/mehrancodes/laravel-harbor/app/Services/Forge/Pipeline/ObtainLetsEncryptCertification.php(28): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
2024-04-12T13:23:09.4397184Z #10 /composer/vendor/illuminate/pipeline/Pipeline.php(184): App\\Services\\Forge\\Pipeline\\ObtainLetsEncryptCertification->__invoke()
2024-04-12T13:23:09.4399123Z #11 /composer/vendor/mehrancodes/laravel-harbor/app/Services/Forge/Pipeline/InstallGitRepository.php(27): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
2024-04-12T13:23:09.4400791Z #12 /composer/vendor/illuminate/pipeline/Pipeline.php(184): App\\Services\\Forge\\Pipeline\\InstallGitRepository->__invoke()
2024-04-12T13:23:09.4402652Z #13 /composer/vendor/mehrancodes/laravel-harbor/app/Services/Forge/Pipeline/CreateDatabase.php(28): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
2024-04-12T13:23:09.4404145Z #14 /composer/vendor/illuminate/pipeline/Pipeline.php(184): App\\Services\\Forge\\Pipeline\\CreateDatabase->__invoke()
2024-04-12T13:23:09.4405788Z #15 /composer/vendor/mehrancodes/laravel-harbor/app/Services/Forge/Pipeline/NginxTemplateSearchReplace.php(31): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
2024-04-12T13:23:09.4407402Z #16 /composer/vendor/illuminate/pipeline/Pipeline.php(184): App\\Services\\Forge\\Pipeline\\NginxTemplateSearchReplace->__invoke()
2024-04-12T13:23:09.4410044Z #17 /composer/vendor/mehrancodes/laravel-harbor/app/Services/Forge/Pipeline/OrCreateNewSite.php(35): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
2024-04-12T13:23:09.4412671Z #18 /composer/vendor/illuminate/pipeline/Pipeline.php(184): App\\Services\\Forge\\Pipeline\\OrCreateNewSite->__invoke()
2024-04-12T13:23:09.4414408Z #19 /composer/vendor/mehrancodes/laravel-harbor/app/Services/Forge/Pipeline/FindSite.php(32): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
2024-04-12T13:23:09.4415846Z #20 /composer/vendor/illuminate/pipeline/Pipeline.php(184): App\\Services\\Forge\\Pipeline\\FindSite->__invoke()
2024-04-12T13:23:09.4417403Z #21 /composer/vendor/mehrancodes/laravel-harbor/app/Services/Forge/Pipeline/FindServer.php(34): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
2024-04-12T13:23:09.4418799Z #22 /composer/vendor/illuminate/pipeline/Pipeline.php(184): App\\Services\\Forge\\Pipeline\\FindServer->__invoke()
2024-04-12T13:23:09.4420073Z #23 /composer/vendor/illuminate/pipeline/Pipeline.php(119): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
2024-04-12T13:23:09.4421475Z #24 /composer/vendor/mehrancodes/laravel-harbor/app/Commands/ProvisionCommand.php(66): Illuminate\\Pipeline\\Pipeline->then()
2024-04-12T13:23:09.4422738Z #25 /composer/vendor/illuminate/container/BoundMethod.php(36): App\\Commands\\ProvisionCommand->handle()
2024-04-12T13:23:09.4449291Z #26 /composer/vendor/illuminate/container/Util.php(41): Illuminate\\Container\\BoundMethod::Illuminate\\Container\\{closure}()
2024-04-12T13:23:09.4450803Z #27 /composer/vendor/illuminate/container/BoundMethod.php(93): Illuminate\\Container\\Util::unwrapIfClosure()
2024-04-12T13:23:09.4451947Z #28 /composer/vendor/illuminate/container/BoundMethod.php(35): Illuminate\\Container\\BoundMethod::callBoundMethod()
2024-04-12T13:23:09.4453069Z #29 /composer/vendor/illuminate/container/Container.php(662): Illuminate\\Container\\BoundMethod::call()
2024-04-12T13:23:09.4454326Z #30 /composer/vendor/illuminate/console/Command.php(211): Illuminate\\Container\\Container->call()
2024-04-12T13:23:09.4455415Z #31 /composer/vendor/symfony/console/Command/Command.php(326): Illuminate\\Console\\Command->execute()
2024-04-12T13:23:09.4456501Z #32 /composer/vendor/illuminate/console/Command.php(180): Symfony\\Component\\Console\\Command\\Command->run()
2024-04-12T13:23:09.4458390Z #33 /composer/vendor/symfony/console/Application.php(1096): Illuminate\\Console\\Command->run()
2024-04-12T13:23:09.4459556Z #34 /composer/vendor/symfony/console/Application.php(324): Symfony\\Component\\Console\\Application->doRunCommand()
2024-04-12T13:23:09.4460826Z #35 /composer/vendor/symfony/console/Application.php(175): Symfony\\Component\\Console\\Application->doRun()
2024-04-12T13:23:09.4462234Z #36 /composer/vendor/laravel-zero/foundation/src/Illuminate/Foundation/Console/Kernel.php(201): Symfony\\Component\\Console\\Application->run()
2024-04-12T13:23:09.4464195Z #37 /composer/vendor/laravel-zero/framework/src/Kernel.php(97): Illuminate\\Foundation\\Console\\Kernel->handle()
2024-04-12T13:23:09.4465409Z #38 /composer/vendor/mehrancodes/laravel-harbor/harbor(35): LaravelZero\\Framework\\Kernel->handle()
2024-04-12T13:23:09.4466417Z #39 /composer/vendor/bin/harbor(119): include('...')
2024-04-12T13:23:09.4466875Z #40 {main}
2024-04-12T13:23:09.4467225Z "} 
2024-04-12T13:23:09.4467452Z 
2024-04-12T13:23:09.4467573Z    Exception 
2024-04-12T13:23:09.4467786Z 
2024-04-12T13:23:09.4467890Z   {
2024-04-12T13:23:09.4468289Z     "message": "Server Error"
2024-04-12T13:23:09.4468619Z }
2024-04-12T13:23:09.4468795Z 
2024-04-12T13:23:09.4469136Z   at /composer/vendor/laravel/forge-sdk/src/MakesHttpRequests.php:125
2024-04-12T13:23:09.4469903Z     121▕                     : null
2024-04-12T13:23:09.4470328Z     122▕             );
2024-04-12T13:23:09.4470658Z     123▕         }
2024-04-12T13:23:09.4471095Z     124▕ 
2024-04-12T13:23:09.4471576Z   ➜ 125▕         throw new Exception((string) $response->getBody());
2024-04-12T13:23:09.4472124Z     126▕     }
2024-04-12T13:23:09.4472514Z     127▕ 
2024-04-12T13:23:09.4472835Z     128▕     /**
2024-04-12T13:23:09.4473305Z     129▕      * Retry the callback or fail after x seconds.
2024-04-12T13:23:09.4473731Z 
2024-04-12T13:23:09.4473735Z 
2024-04-12T13:23:09.4483274Z ##[error]Process completed with exit code 1.
2024-04-12T13:23:09.4498245Z ##[debug]Finishing: Start Provisioning
2024-04-12T13:23:09.4517074Z ##[debug]Evaluating condition for step: 'Stop containers'
2024-04-12T13:23:09.4519677Z ##[debug]Evaluating: always()
2024-04-12T13:23:09.4520395Z ##[debug]Evaluating always:
2024-04-12T13:23:09.4521407Z ##[debug]=> true
2024-04-12T13:23:09.4522186Z ##[debug]Result: true
2024-04-12T13:23:09.4523091Z ##[debug]Starting: Stop containers
2024-04-12T13:23:09.4554565Z Stop and remove container: a4b1c5e61e744335aaf2387a9e8d1457_kirschbaumdevelopmentlaraveltestrunner83_9f0f06
2024-04-12T13:23:09.4560004Z ##[command]/usr/bin/docker rm --force c5b0276a75a8bb2849a46fccf3b73b5df39f8da1d05d54e2917015afa5af7ed3
2024-04-12T13:23:09.9295554Z c5b0276a75a8bb2849a46fccf3b73b5df39f8da1d05d54e2917015afa5af7ed3
2024-04-12T13:23:09.9332676Z Remove container network: github_network_da9f8e9124b149dc95ad04d22529e2d8
2024-04-12T13:23:09.9339075Z ##[command]/usr/bin/docker network rm github_network_da9f8e9124b149dc95ad04d22529e2d8
2024-04-12T13:23:10.1322470Z github_network_da9f8e9124b149dc95ad04d22529e2d8
2024-04-12T13:23:10.1337619Z ##[debug]Finishing: Stop containers
2024-04-12T13:23:10.1494893Z ##[debug]Starting: Complete job
2024-04-12T13:23:10.1497494Z Uploading runner diagnostic logs
2024-04-12T13:23:10.1550118Z ##[debug]Starting diagnostic file upload.
2024-04-12T13:23:10.1550919Z ##[debug]Setting up diagnostic log folders.
2024-04-12T13:23:10.1555317Z ##[debug]Creating diagnostic log files folder.
2024-04-12T13:23:10.1575862Z ##[debug]Copying 2 worker diagnostic logs.
2024-04-12T13:23:10.1595130Z ##[debug]Copying 1 runner diagnostic logs.
2024-04-12T13:23:10.1596736Z ##[debug]Zipping diagnostic files.
2024-04-12T13:23:10.1654488Z ##[debug]Uploading diagnostic metadata file.
2024-04-12T13:23:10.1679153Z ##[debug]Diagnostic file upload complete.
2024-04-12T13:23:10.1680218Z Completed runner diagnostic log upload
2024-04-12T13:23:10.1680694Z Cleaning up orphan processes
2024-04-12T13:23:10.2069431Z ##[debug]Finishing: Complete job
2024-04-12T13:23:10.2198956Z ##[debug]Finishing: harbor-provision
```

[Full log archive](https://github.com/mehrancodes/laravel-harbor/files/14999539/logs_22700753169.zip)

--- a/app/Actions/MergeEnvironmentVariables.php
+++ b/app/Actions/MergeEnvironmentVariables.php
@@ -40,21 +40,27 @@ class MergeEnvironmentVariables
 
     protected function searchReplaceExistingVariables(string $source, array &$newVariables): string
     {
+        // Determine the separator based on the source string
         $separator = Str::contains($source, ';') ? ';' : "\n";
         $output = '';
 
         foreach (explode($separator, $source) as $variable) {
+            // If the variable is empty, add a newline to the output
             if (empty($variable)) {
                 $output .= "\n";
+                continue;
+            }
 
+            if (Str::contains($variable, '#')) {
+                $output .= "$variable\n";
                 continue;
             }
 
             [$key, $value] = explode('=', $variable, 2);
 
+            // If the key is empty, issue a warning and skip
             if (empty($key)) {
                 $this->warning("No key found for the assigned value \"$value\" inside your environment variables! Make sure to remove it.");
-
                 continue;
             }
 

--- a/tests/Feature/Actions/MergeEnvironmentVariablesTest.php
+++ b/tests/Feature/Actions/MergeEnvironmentVariablesTest.php
@@ -67,4 +67,11 @@ it('it can merge custom variables with source environment variables securely', f
             ],
             'expected' => "APP_NAME=Project Name\n",
         ],
+        [
+            'actual' => [
+                'source' => "APP_NAME=Laravel\n# Here be dragons\nAPP_ENV=local\n",
+                'content' => [],
+            ],
+            'expected' => "APP_NAME=Laravel\n# Here be dragons\nAPP_ENV=local\n\n",
+        ],
     ]);

